src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamNegotiatedCipherSuiteTest.cs

```diff
@@ -20,9 +20,8 @@ namespace System.Net.Security.Tests
 
     public class NegotiatedCipherSuiteTest
     {
-#pragma warning disable CS0618 // Ssl2 and Ssl3 are obsolete
-        public const SslProtocols AllProtocols =
-            SslProtocols.Ssl2 | SslProtocols.Ssl3 |
+#pragma warning disable CS0618 // Ssl3 are obsolete
+        public const SslProtocols AllProtocols = SslProtocols.Ssl3 |
```

Comment:
Why was Ssl2 removed?

Reply:
Ssl2 does not have the framing protocol So determining length and type is somewhat more complicated. And I did not want to bring that to the TLS parser. With Ssl3 being deprecated and Ssl2 not supported by any of the current Ones I felt that would be ok. 
rfc6176 from 2011 prohibts use of Ssl2.

Reply:
Does that mean your change removes Ssl2 support from SslStream?

Reply:
yes, I mention that in description. I can mark this as breaking change to make that more visible. I can also split this or roll back that part and I keep old processing around.

Reply:
I think we need to have a broader conversation about that.  On the one hand, we don't want anyone using SSL2; it's insecure, deprecated, etc.  On the other hand, we know we've gotten pushback in the past because of apps talking to, e.g., services and devices that are out of their control.  For the most part then we've left it up to whether the underlying operating system / platform supports it.  This would be explicitly disabling such support even when it's enabled in the OS.

Reply:
We shouldn't remove it completely yet from .NET 5. We still seem to run into customer issues where they have some very old device that they can't change but it talks some legacy protocol. We don't want to prevent customers from using .NET 5 due to this.

At some point, we probably would drop support from it. But I think .NET 5 is not the release to do that.

It would be interesting to get other data points such as does OpenSsl still support SSL2 in some fashion?

Reply:
ok, I'll split it. maybe open new issue about retiring Sslv2 to be more explicit?

Reply:
> does OpenSsl still support SSL2 in some fashion?

They deleted all the SSLv2 code for OpenSSL 1.1.0.   So Ubuntu 16.04 can probably still do it (OpenSSL 1.0.2), but in another year I don't think there will be any supported Linux configurations that can do SSLv2.

Reply:
While Ubuntu16 has -ssl2, attempt to use it throws error so as forced run of ClientAsyncAuthenticate_Ssl2WithSelf_Success. It seems like `!PlatformDetection.IsWindows10Version1607OrGreater` is only viable option. 
I created #35942 to track this and rolled-back Ssl2 changes. 
I agree that it should not be side-effect of this change (even if convenient in this case)